src/fu-tool.c

```diff
@@ -1375,13 +1376,32 @@ fu_util_activate (FuUtilPrivate *priv, gchar **values, GError **error)
 	if (!fu_util_start_engine (priv, FU_ENGINE_LOAD_FLAG_READONLY_FS, error))
 		return FALSE;
 
+	engine_devices = fu_engine_get_devices (priv->engine, error);
+	if (engine_devices == NULL)
+		return FALSE;
+	if (engine_devices->len == 0) {
+		g_printerr ("No devices to activate\n");
+		return TRUE;
+	}
+
 	/* activate anything with _NEEDS_ACTIVATION */
-	for (guint i = 0; i < devices->len; i++) {
-		FuDevice *device = g_ptr_array_index (devices, i);
+	for (guint i = 0; i < history_devices->len; i++) {
+		FuDevice *device = g_ptr_array_index (history_devices, i);
 		if (!fu_util_filter_device (priv, FWUPD_DEVICE (device)))
 			continue;
 		if (!fu_device_has_flag (device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION))
 			continue;
+		/* make sure that it's plugged in */
+		has_pending = FALSE;
+		for (guint j = 0; j < engine_devices->len; j++) {
+			FuDevice *engine_device = g_ptr_array_index (engine_devices, j);
+			if (g_strcmp0 (fu_device_get_id (engine_device), fu_device_get_id (device)) == 0)
+				has_pending = TRUE;
+		}
+		if (!has_pending) {
+			g_debug ("unable to find %s (%s)", fu_device_get_name (device), fu_device_get_id (device));
```

Comment:
don't we normally use square brackets for the ID?

Reply:
fixed